docs(integration/providers):add Truefoundry AI Gateway (#2)
rishiraj-tf/langchain: docs/docs/integrations/providers/truefoundry.mdx

# TrueFoundry

> TrueFoundry provides an enterprise-ready [AI Gateway](https://www.truefoundry.com/ai-gateway) to provide governance and observability to agentic frameworks like LangChain. TrueFoundry AI Gateway serves as a unified interface for LLM access, providing:

Reviewer: ```suggestion
TrueFoundry provides an enterprise-ready [AI Gateway](https://www.truefoundry.com/ai-gateway) which can integrate with agentic frameworks like LangChain and provides governance and observability for your AI Applications. TrueFoundry AI Gateway serves as a unified interface for LLM access, providing:
```

```python
from langchain_openai import ChatOpenAI

llm = ChatOpenAI(
    api_key=TRUEFOUNDRY_API_KEY,
    base_url=TRUEFOUNDRY_GATEWAY_BASE_URL,

>     model="openai-main/gpt-4o"   #similarly you can call any model from any model provider

Reviewer: ```suggestion
    api_key="TRUEFOUNDRY_API_KEY",
    base_url="TRUEFOUNDRY_GATEWAY_BASE_URL",
    model="openai-main/gpt-4o"   #similarly you can call any model from any model provider like anthropic/claude-4
```

result = app.invoke({"messages": [HumanMessage(content="Hello!")]})
```

## Observability and Governance

![TrueFoundry metrics](/img/gateway-metrics.png)

With the Metrics Dashboard, you can monitor and analyze:

> - **Performance Metrics**: Track key latency metrics like Request Latency, Time to First Token (TTFS), and Inter-Token Latency (ITL) with P99, P90, and P50 percentiles
> - **Cost and Token Usage**: Gain visibility into your application's costs with detailed breakdowns of input/output tokens and the associated expenses for each model
> - **Usage Patterns**: Understand how your application is being used with detailed analytics on user activity, model distribution, and team-based usage
> - **Rate Limiting & Load Balancing**: Configure limits, distribute traffic across models, and set up fallbacks

Reviewer: ```suggestion
- **Performance Metrics**: Track key latency metrics like Request Latency, Time to First Token (TTFS), and Inter-Token Latency (ITL) with P99, P90, and P50 percentiles
- **Cost and Token Usage**: Gain visibility into your application's costs with detailed breakdowns of input/output tokens and the associated expenses for each model
- **Rate Limiting & Load Balancing**: Configure limits, distribute traffic across models, and set up fallbacks
- **Security and Governance**:  <guardrail authZ>
```
